ansible/roles/fanout/templates/sonic_deploy_202405.j2

```diff
@@ -18,6 +18,16 @@
             "mtu": "9100",
     {% if fanout_hwsku == "Arista-720DT-G48S4" and (fanout_port_config[port_name]['lanes'] | int) <= 24 %}
             "autoneg": "on",
+    {% else %}
+        {% if port_name in device_conn[inventory_hostname] and 'autoneg' in device_conn[inventory_hostname][port_name] %}
+            "autoneg": "{{ device_conn[inventory_hostname][port_name]['autoneg'] }}",
+        {% else %}
+            {% if "Mellanox-SN5" in fanout_hwsku or "Mellanox-SN4" in fanout_hwsku %}
```

Comment:
We also have Mellanox-SN4600, which doesn't require AN. So better to check a list of HWSKU ?

Reply:
Could you please provide a list of HWSKU to me?

Reply:
You may find all the HWSKU here https://github.com/sonic-net/sonic-buildimage/tree/master/device/mellanox/x86_64-mlnx_msn4600c-r0

Reply:
if we already have done the check for autoneg = 'on'  in the *links.csv, do we really require a hwsku check ?

Reply:
This is for the default scenario. Because the *links.csv didn't include all ports.
For example, a SKU has 64 ports, but there is only 32 ports in *links.csv.

Reply:
I suggest checking the port speed. Enable AN if port speed >= 200Gb.